### DISCUSSION TITLE
Help manage MongoDB connection pools (Vercel + MongoDB sucks)

### DISCUSSION THREAD

**Discussion by juliuslissau** [vercel/community#4297] [Idea]
### Question

Using MongoDB Atlas and Next.js API (Vercel serverless functions) is a pain. Connection pooling doesn't work so connections numbers get super high and requires expensive MongoDB Atlas tiers. 

It would be amazing if Vercel could create (or at least guide) to some sort of solution for this problem. For example like with Vercel have done with PostgreSQL connection pool. Or in another way.

I appreciate this article https://vercel.com/guides/connection-pooling-with-serverless-functions but it is not enough.

**Comment by Hagbarths** [comment]
I completely agree. This have been a pain in "our ass" too, we have spend so many hours trying to fix this, but seems like there is no solution.

**Comment by renenielsendk** [comment]
+1
We are getting alerts weekly on this, and currently our best solution would be to completely migtate db stack, which is not feasible.

**Comment by aldosch** [comment]
Hi everyone.
We really appreciate your patience, I understand this was raised a long time ago.

[There is a related discussion here](https://github.com/vercel/vercel/discussions/11178#discussioncomment-9503785.). We encourage you to follow [**NODE-6179**](https://jira.mongodb.org/browse/NODE-6179) which is a bug report submitted to MongoDB covering the same connection timeout problem. If you could share any additional context or troubleshooting attempts there it would be very helpful.

I agree that our documentation could be improved on this topic. We're hopeful that once the root cause of the timeouts has been identified we can document and share the solution.

Thanks,
Aldo
